src/realm/obj.cpp

```diff
@@ -222,6 +222,37 @@ CollectionListPtr CollectionList::get_collection_list(size_t ndx) const
     return CollectionList::create(const_cast<CollectionList*>(this)->shared_from_this(), m_col_key, index, coll_type);
 }
 
+void CollectionList::remove(size_t ndx)
+{
+    REALM_ASSERT(m_key_type == type_Int);
+    auto int_keys = static_cast<BPlusTree<Int>*>(m_keys.get());
+    if (ndx < int_keys->size()) {
+        int_keys->erase(ndx);
+        auto ref = m_refs.get(ndx);
+        Array::destroy_deep(to_ref(ref), *m_alloc);
+        m_refs.erase(ndx);
+        return;
+    }
+    throw InvalidArgument("Index out of bound");
```

Comment:
I would prefer `throw OutOfBounds("CollectionList::remove", ndx, sz);`
you can have `size_t sz = int_keys->size();` further up.